packages/core/src/core/fetch.ts

```
@@ -307,13 +307,12 @@ const calcFetchChanges = async (
     workspaceElements,
     mergedServiceElements,
   ), 'finished to calculate service-workspace changes')
-
   const serviceElementsSource = elementSource.createInMemoryElementSource(serviceElements)
 
   return awu(serviceChanges)
     .flatMap(toFetchChanges(pendingChanges, workspaceToServiceChanges))
     .flatMap(toChangesWithPath(
-      async name => collections.array.makeArray(await serviceElementsSource.get(name))
+      async name => [await serviceElementsSource.get(name)].filter(values.isDefined)
```

nit:  rename `name` -> `elemID`

Also, is the case that the elementsSource returns an undefined here really a viable case?

---

Its not a viable case - but since get can return undefined we need to filter to make TS compile (Its either this, or a cast)

---

If it's not viable, should we be silent about this case?

---

I mean, we could throw an error I guess to make things more explicit - but all we will gain from that is lower test coverage :) So I think its OK to just filter (or cast, if you prefer. We can also cast)

---

Didn't you hear of `Thou shall not cast`??

If you are 100% certain it can't happen then OK, I am just afraid of the bug hiding here :)